### PR TITLE
[SHACK-250] Renaming the chef-run gem to chef-apply

### PR DESCRIPTION
**Pull request by tyler-ball**
Splitting the `chef-run` gem into its own repo and renaming it to the `chef-apply` gem. It still keeps an executable name of `chef-run` for now

### CODE REVIEW

**Comment by tyler-ball** [comment]
Need a review from @chef/expeditor-maintainers

File: Gemfile [hunk 1]
```diff
@@ -18,11 +18,6 @@
 source "https://rubygems.org"
 gemspec
 
-# TODO when chef-dk 3.0 is released to Rubygems as 3.0 we can get rid of this
-gem "chef-dk", git: "https://github.com/chef/chef-dk.git", branch: "master"
-
-gem "train", git: "https://github.com/chef/train.git", branch: "v1.4.6"
```
Comment (marcparadise) [inline Gemfile]: Concerned with dropping this pin - last time we tried (1.4.8?) we had health check failures in the windows build iirc.

File: .expeditor/update_version.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+#
+# After a PR merge, Chef Expeditor will bump the PATCH version in the VERSION file.
+# It then executes this file to update any other files/components with that new version.
+#
+
+set -evx
+
+sed -i -r "s/VERSION = \".*\"/VERSION = \"$(cat VERSION)\"/"  lib/chef_apply/version.rb
```
Comment (marcparadise) [inline .expeditor/update_version.sh]: Why the change to an underscored chef_apply instead of hyphenated?
